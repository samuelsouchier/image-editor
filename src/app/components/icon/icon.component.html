<div class="icon" [innerHTML]="icon"></div>
