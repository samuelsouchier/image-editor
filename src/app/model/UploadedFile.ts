export interface UploadedFile {
  id: string,
  file: File,
}
