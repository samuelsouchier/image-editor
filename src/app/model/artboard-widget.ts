export interface ArtboardWidgetPosition {
  x: number,
  y: number,
}

export interface ArtboardWidgetImage {
  id: string;
  src: string;
  name: string;
  position: ArtboardWidgetPosition;
  height: number;
}
