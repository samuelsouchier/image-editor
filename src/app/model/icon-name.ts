import { blackWhiteIcon } from '../../assets/icons/black-white';
import { blurIcon } from '../../assets/icons/blur';
import { brightnessIcon } from '../../assets/icons/brightness';
import { clearIcon } from '../../assets/icons/clear';
import { contrastIcon } from '../../assets/icons/contrast';
import { cropIcon } from '../../assets/icons/crop';
import { imageIcon } from '../../assets/icons/image';
import { maximizeIcon } from '../../assets/icons/maximize';
import { moveIcon } from '../../assets/icons/move';
import { rotateBackIcon } from '../../assets/icons/rotate-back';
import { rotateForwardIcon } from '../../assets/icons/rotate-forward';
import { sepiaIcon } from '../../assets/icons/sepia';
import { trashIcon } from '../../assets/icons/trash';
import { typeIcon } from '../../assets/icons/type';
import { zoomInIcon } from '../../assets/icons/zoom-in';
import { zoomOutIcon } from '../../assets/icons/zoom-out';

export type IconName =
  'image'
  | 'trash'
  | 'crop'
  | 'type'
  | 'maximize'
  | 'move'
  | 'clear'
  | 'rotateBack'
  | 'rotateForward'
  | 'brightness'
  | 'blur'
  | 'contrast'
  | 'blackWhite'
  | 'sepia'
  | 'zoomIn'
  | 'zoomOut'
  ;
export const iconByNameMapper: Record<IconName, string> = {
  'image': imageIcon,
  'trash': trashIcon,
  'type': typeIcon,
  'crop': cropIcon,
  'maximize': maximizeIcon,
  'move': moveIcon,
  'clear': clearIcon,
  'rotateBack': rotateBackIcon,
  'rotateForward': rotateForwardIcon,
  'brightness': brightnessIcon,
  'blur': blurIcon,
  'contrast': contrastIcon,
  'blackWhite': blackWhiteIcon,
  'sepia': sepiaIcon,
  'zoomIn': zoomInIcon,
  'zoomOut': zoomOutIcon,
};
