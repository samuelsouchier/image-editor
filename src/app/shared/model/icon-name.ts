import { cropIcon } from '../../../assets/icons/crop';
import { imageIcon } from '../../../assets/icons/image';
import { trashIcon } from '../../../assets/icons/trash';
import { typeIcon } from '../../../assets/icons/type';

export type IconName = 'image' | 'trash' | 'crop' | 'type';
export const iconByNameMapper: Record<IconName, string> = {
  'image': imageIcon,
  'trash': trashIcon,
  'type': typeIcon,
  'crop': cropIcon,
};
