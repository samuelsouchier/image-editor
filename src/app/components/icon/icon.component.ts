import { CommonModule } from '@angular/common';
import { Component, Input, OnInit } from '@angular/core';
import { DomSanitizer, SafeHtml } from '@angular/platform-browser';
import { iconByNameMapper, IconName } from '../../model/icon-name';

@Component({
  selector: 'app-icon',
  standalone: true,
  imports: [CommonModule],
  templateUrl: './icon.component.html',
  styleUrls: ['./icon.component.scss']
})
export class IconComponent implements OnInit {
  @Input({required: true}) iconName: IconName;
  icon: SafeHtml;

  constructor(private sanitizer: DomSanitizer) {
  }
  ngOnInit() {
    this.icon = this.sanitizer.bypassSecurityTrustHtml(iconByNameMapper[this.iconName]);
  }
}
