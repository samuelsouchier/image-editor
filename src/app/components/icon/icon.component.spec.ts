import { createComponentFactory, Spectator } from '@ngneat/spectator/jest';
import { iconByNameMapper, IconName } from '../../model/icon-name';

import { IconComponent } from './icon.component';

describe(IconComponent.name, () => {
  let spectator: Spectator<IconComponent>;
  const createComponent = createComponentFactory({
    component: IconComponent,
    shallow: true,
  });

  for (const key in Object.keys(iconByNameMapper)) {
    const iconName = Object.keys(iconByNameMapper)[key] as IconName;
    it(`should match snapshot for ${iconName} icon`, () => {
      spectator = createComponent({props: {iconName}});

      expect(spectator.fixture).toMatchSnapshot();
    });
  }
});
