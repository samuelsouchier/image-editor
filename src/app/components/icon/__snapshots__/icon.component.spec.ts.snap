// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`IconComponent should match snapshot for blackWhite icon 1`] = `
<app-icon
  icon={[Function SafeHtmlImpl]}
  iconName={[Function String]}
  sanitizer={[Function _DomSanitizerImpl]}
>
  <div
    class="icon"
  >
    <svg
      class="feather feather-copy"
      fill="none"
      height="24"
      stroke="currentColor"
      stroke-linecap="round"
      stroke-linejoin="round"
      stroke-width="2"
      viewBox="0 0 24 24"
      width="24"
      xmlns="http://www.w3.org/2000/svg"
    >
      <rect
        height="13"
        rx="2"
        ry="2"
        width="13"
        x="9"
        y="9"
      />
      <path
        d="M5 15H4a2 2 0 0 1-2-2V4a2 2 0 0 1 2-2h9a2 2 0 0 1 2 2v1"
      />
    </svg>
  </div>
</app-icon>
`;

exports[`IconComponent should match snapshot for blur icon 1`] = `
<app-icon
  icon={[Function SafeHtmlImpl]}
  iconName={[Function String]}
  sanitizer={[Function _DomSanitizerImpl]}
>
  <div
    class="icon"
  >
    <svg
      class="feather feather-droplet"
      fill="none"
      height="24"
      stroke="currentColor"
      stroke-linecap="round"
      stroke-linejoin="round"
      stroke-width="2"
      viewBox="0 0 24 24"
      width="24"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M12 2.69l5.66 5.66a8 8 0 1 1-11.31 0z"
      />
    </svg>
  </div>
</app-icon>
`;

exports[`IconComponent should match snapshot for brightness icon 1`] = `
<app-icon
  icon={[Function SafeHtmlImpl]}
  iconName={[Function String]}
  sanitizer={[Function _DomSanitizerImpl]}
>
  <div
    class="icon"
  >
    <svg
      class="feather feather-sun"
      fill="none"
      height="24"
      stroke="currentColor"
      stroke-linecap="round"
      stroke-linejoin="round"
      stroke-width="2"
      viewBox="0 0 24 24"
      width="24"
      xmlns="http://www.w3.org/2000/svg"
    >
      <circle
        cx="12"
        cy="12"
        r="5"
      />
      <line
        x1="12"
        x2="12"
        y1="1"
        y2="3"
      />
      <line
        x1="12"
        x2="12"
        y1="21"
        y2="23"
      />
      <line
        x1="4.22"
        x2="5.64"
        y1="4.22"
        y2="5.64"
      />
      <line
        x1="18.36"
        x2="19.78"
        y1="18.36"
        y2="19.78"
      />
      <line
        x1="1"
        x2="3"
        y1="12"
        y2="12"
      />
      <line
        x1="21"
        x2="23"
        y1="12"
        y2="12"
      />
      <line
        x1="4.22"
        x2="5.64"
        y1="19.78"
        y2="18.36"
      />
      <line
        x1="18.36"
        x2="19.78"
        y1="5.64"
        y2="4.22"
      />
    </svg>
  </div>
</app-icon>
`;

exports[`IconComponent should match snapshot for clear icon 1`] = `
<app-icon
  icon={[Function SafeHtmlImpl]}
  iconName={[Function String]}
  sanitizer={[Function _DomSanitizerImpl]}
>
  <div
    class="icon"
  >
    <svg
      class="feather feather-x-circle"
      fill="none"
      height="24"
      stroke="currentColor"
      stroke-linecap="round"
      stroke-linejoin="round"
      stroke-width="2"
      viewBox="0 0 24 24"
      width="24"
      xmlns="http://www.w3.org/2000/svg"
    >
      <circle
        cx="12"
        cy="12"
        r="10"
      />
      <line
        x1="15"
        x2="9"
        y1="9"
        y2="15"
      />
      <line
        x1="9"
        x2="15"
        y1="9"
        y2="15"
      />
    </svg>
  </div>
</app-icon>
`;

exports[`IconComponent should match snapshot for contrast icon 1`] = `
<app-icon
  icon={[Function SafeHtmlImpl]}
  iconName={[Function String]}
  sanitizer={[Function _DomSanitizerImpl]}
>
  <div
    class="icon"
  >
    <svg
      class="feather feather-moon"
      fill="none"
      height="24"
      stroke="currentColor"
      stroke-linecap="round"
      stroke-linejoin="round"
      stroke-width="2"
      viewBox="0 0 24 24"
      width="24"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M21 12.79A9 9 0 1 1 11.21 3 7 7 0 0 0 21 12.79z"
      />
    </svg>
  </div>
</app-icon>
`;

exports[`IconComponent should match snapshot for crop icon 1`] = `
<app-icon
  icon={[Function SafeHtmlImpl]}
  iconName={[Function String]}
  sanitizer={[Function _DomSanitizerImpl]}
>
  <div
    class="icon"
  >
    <svg
      class="feather feather-crop"
      fill="none"
      height="24"
      stroke="currentColor"
      stroke-linecap="round"
      stroke-linejoin="round"
      stroke-width="2"
      viewBox="0 0 24 24"
      width="24"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M6.13 1L6 16a2 2 0 0 0 2 2h15"
      />
      <path
        d="M1 6.13L16 6a2 2 0 0 1 2 2v15"
      />
    </svg>
  </div>
</app-icon>
`;

exports[`IconComponent should match snapshot for image icon 1`] = `
<app-icon
  icon={[Function SafeHtmlImpl]}
  iconName={[Function String]}
  sanitizer={[Function _DomSanitizerImpl]}
>
  <div
    class="icon"
  >
    <svg
      class="feather feather-image"
      fill="none"
      height="24"
      stroke="currentColor"
      stroke-linecap="round"
      stroke-linejoin="round"
      stroke-width="2"
      viewBox="0 0 24 24"
      width="24"
      xmlns="http://www.w3.org/2000/svg"
    >
      <rect
        height="18"
        rx="2"
        ry="2"
        width="18"
        x="3"
        y="3"
      />
      <circle
        cx="8.5"
        cy="8.5"
        r="1.5"
      />
      <polyline
        points="21 15 16 10 5 21"
      />
    </svg>
  </div>
</app-icon>
`;

exports[`IconComponent should match snapshot for maximize icon 1`] = `
<app-icon
  icon={[Function SafeHtmlImpl]}
  iconName={[Function String]}
  sanitizer={[Function _DomSanitizerImpl]}
>
  <div
    class="icon"
  >
    <svg
      class="feather feather-maximize-2"
      fill="none"
      height="24"
      stroke="currentColor"
      stroke-linecap="round"
      stroke-linejoin="round"
      stroke-width="2"
      viewBox="0 0 24 24"
      width="24"
      xmlns="http://www.w3.org/2000/svg"
    >
      <polyline
        points="15 3 21 3 21 9"
      />
      <polyline
        points="9 21 3 21 3 15"
      />
      <line
        x1="21"
        x2="14"
        y1="3"
        y2="10"
      />
      <line
        x1="3"
        x2="10"
        y1="21"
        y2="14"
      />
    </svg>
  </div>
</app-icon>
`;

exports[`IconComponent should match snapshot for move icon 1`] = `
<app-icon
  icon={[Function SafeHtmlImpl]}
  iconName={[Function String]}
  sanitizer={[Function _DomSanitizerImpl]}
>
  <div
    class="icon"
  >
    <svg
      class="feather feather-move"
      fill="none"
      height="24"
      stroke="currentColor"
      stroke-linecap="round"
      stroke-linejoin="round"
      stroke-width="2"
      viewBox="0 0 24 24"
      width="24"
      xmlns="http://www.w3.org/2000/svg"
    >
      <polyline
        points="5 9 2 12 5 15"
      />
      <polyline
        points="9 5 12 2 15 5"
      />
      <polyline
        points="15 19 12 22 9 19"
      />
      <polyline
        points="19 9 22 12 19 15"
      />
      <line
        x1="2"
        x2="22"
        y1="12"
        y2="12"
      />
      <line
        x1="12"
        x2="12"
        y1="2"
        y2="22"
      />
    </svg>
  </div>
</app-icon>
`;

exports[`IconComponent should match snapshot for rotateBack icon 1`] = `
<app-icon
  icon={[Function SafeHtmlImpl]}
  iconName={[Function String]}
  sanitizer={[Function _DomSanitizerImpl]}
>
  <div
    class="icon"
  >
    <svg
      class="feather feather-corner-left-down"
      fill="none"
      height="24"
      stroke="currentColor"
      stroke-linecap="round"
      stroke-linejoin="round"
      stroke-width="2"
      viewBox="0 0 24 24"
      width="24"
      xmlns="http://www.w3.org/2000/svg"
    >
      <polyline
        points="14 15 9 20 4 15"
      />
      <path
        d="M20 4h-7a4 4 0 0 0-4 4v12"
      />
    </svg>
  </div>
</app-icon>
`;

exports[`IconComponent should match snapshot for rotateForward icon 1`] = `
<app-icon
  icon={[Function SafeHtmlImpl]}
  iconName={[Function String]}
  sanitizer={[Function _DomSanitizerImpl]}
>
  <div
    class="icon"
  >
    <svg
      class="feather feather-corner-right-down"
      fill="none"
      height="24"
      stroke="currentColor"
      stroke-linecap="round"
      stroke-linejoin="round"
      stroke-width="2"
      viewBox="0 0 24 24"
      width="24"
      xmlns="http://www.w3.org/2000/svg"
    >
      <polyline
        points="10 15 15 20 20 15"
      />
      <path
        d="M4 4h7a4 4 0 0 1 4 4v12"
      />
    </svg>
  </div>
</app-icon>
`;

exports[`IconComponent should match snapshot for sepia icon 1`] = `
<app-icon
  icon={[Function SafeHtmlImpl]}
  iconName={[Function String]}
  sanitizer={[Function _DomSanitizerImpl]}
>
  <div
    class="icon"
  >
    <svg
      class="feather feather-camera"
      fill="none"
      height="24"
      stroke="currentColor"
      stroke-linecap="round"
      stroke-linejoin="round"
      stroke-width="2"
      viewBox="0 0 24 24"
      width="24"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M23 19a2 2 0 0 1-2 2H3a2 2 0 0 1-2-2V8a2 2 0 0 1 2-2h4l2-3h6l2 3h4a2 2 0 0 1 2 2z"
      />
      <circle
        cx="12"
        cy="13"
        r="4"
      />
    </svg>
  </div>
</app-icon>
`;

exports[`IconComponent should match snapshot for trash icon 1`] = `
<app-icon
  icon={[Function SafeHtmlImpl]}
  iconName={[Function String]}
  sanitizer={[Function _DomSanitizerImpl]}
>
  <div
    class="icon"
  >
    <svg
      class="feather feather-trash"
      fill="none"
      height="24"
      stroke="currentColor"
      stroke-linecap="round"
      stroke-linejoin="round"
      stroke-width="2"
      viewBox="0 0 24 24"
      width="24"
      xmlns="http://www.w3.org/2000/svg"
    >
      <polyline
        points="3 6 5 6 21 6"
      />
      <path
        d="M19 6v14a2 2 0 0 1-2 2H7a2 2 0 0 1-2-2V6m3 0V4a2 2 0 0 1 2-2h4a2 2 0 0 1 2 2v2"
      />
    </svg>
  </div>
</app-icon>
`;

exports[`IconComponent should match snapshot for type icon 1`] = `
<app-icon
  icon={[Function SafeHtmlImpl]}
  iconName={[Function String]}
  sanitizer={[Function _DomSanitizerImpl]}
>
  <div
    class="icon"
  >
    <svg
      class="feather feather-type"
      fill="none"
      height="24"
      stroke="currentColor"
      stroke-linecap="round"
      stroke-linejoin="round"
      stroke-width="2"
      viewBox="0 0 24 24"
      width="24"
      xmlns="http://www.w3.org/2000/svg"
    >
      <polyline
        points="4 7 4 4 20 4 20 7"
      />
      <line
        x1="9"
        x2="15"
        y1="20"
        y2="20"
      />
      <line
        x1="12"
        x2="12"
        y1="4"
        y2="20"
      />
    </svg>
  </div>
</app-icon>
`;

exports[`IconComponent should match snapshot for zoomIn icon 1`] = `
<app-icon
  icon={[Function SafeHtmlImpl]}
  iconName={[Function String]}
  sanitizer={[Function _DomSanitizerImpl]}
>
  <div
    class="icon"
  >
    <svg
      class="feather feather-zoom-in"
      fill="none"
      height="24"
      stroke="currentColor"
      stroke-linecap="round"
      stroke-linejoin="round"
      stroke-width="2"
      viewBox="0 0 24 24"
      width="24"
      xmlns="http://www.w3.org/2000/svg"
    >
      <circle
        cx="11"
        cy="11"
        r="8"
      />
      <line
        x1="21"
        x2="16.65"
        y1="21"
        y2="16.65"
      />
      <line
        x1="11"
        x2="11"
        y1="8"
        y2="14"
      />
      <line
        x1="8"
        x2="14"
        y1="11"
        y2="11"
      />
    </svg>
  </div>
</app-icon>
`;

exports[`IconComponent should match snapshot for zoomOut icon 1`] = `
<app-icon
  icon={[Function SafeHtmlImpl]}
  iconName={[Function String]}
  sanitizer={[Function _DomSanitizerImpl]}
>
  <div
    class="icon"
  >
    <svg
      class="feather feather-zoom-out"
      fill="none"
      height="24"
      stroke="currentColor"
      stroke-linecap="round"
      stroke-linejoin="round"
      stroke-width="2"
      viewBox="0 0 24 24"
      width="24"
      xmlns="http://www.w3.org/2000/svg"
    >
      <circle
        cx="11"
        cy="11"
        r="8"
      />
      <line
        x1="21"
        x2="16.65"
        y1="21"
        y2="16.65"
      />
      <line
        x1="8"
        x2="14"
        y1="11"
        y2="11"
      />
    </svg>
  </div>
</app-icon>
`;
